### PR TITLE
Add CAP support

### PR DESCRIPTION
**Pull request by vanosg**

### CODE REVIEW

File: src/mod/server.mod/server.c [hunk 1]
```diff
@@ -1817,6 +1817,8 @@ static void server_report(int idx, int details)
   if (hq.tot)
     dprintf(idx, "    %s %d%% (%d msgs)\n", IRC_HELPQUEUE,
             (int) ((float) (hq.tot * 100.0) / (float) maxqmsg), (int) hq.tot);
+  if (cap.negotiated)
```
Comment (michaelortmann) [inline src/mod/server.mod/server.c]: cap.negotiated' will always evaluate to 'true'.

File: src/eggdrop.h [hunk 2]
```diff
@@ -59,6 +59,7 @@
 #define UHOSTMAX    291 + NICKMAX /* 32 (ident) + 3 (\0, !, @) + NICKMAX */
 #define DIRMAX      512           /* paranoia                            */
 #define LOGLINEMAX  767           /* for misc.c/putlog() <cybah>         */
+#define CAPMAX      1024           /* Arbitrary CAP list string max      */
```
Comment (michaelortmann) [inline src/eggdrop.h]: intendation

File: src/mod/server.mod/tclserv.c [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /*
  * tclserv.c -- part of server.mod
- */
-/*
+ *
+ *
```
Comment (michaelortmann) [inline src/mod/server.mod/tclserv.c]: we dont need double empty line

File: src/mod/server.mod/servmsg.c [hunk 4]
```diff
@@ -1148,6 +1152,47 @@ static int got465(char *from, char *msg)
...
+}
+
+static int got421(char *from, char *msg) {
+  newsplit(&msg);
+  putlog(LOG_SERV, "*", "%s reported an error: %s", from, msg);
+
+  return 1;
+}
+
+static int gotcap(char *from, char *msg) {
```
Comment (michaelortmann) [inline src/mod/server.mod/servmsg.c:1204]: eggdrop code style is: "{" at new line. not only here, but other functions also.

File: src/mod/server.mod/servmsg.c [hunk 5]
```diff
@@ -1148,6 +1152,47 @@ static int got465(char *from, char *msg)
...
+    putlog(LOG_MISC, "*", "%s supports CAP sub-commands: %s", from, msg);
+    strlcpy(cap.supported, msg, sizeof cap.supported);
+  }
+  else if (!strcmp(cmd, "LIST")) {
+    putlog(LOG_MISC, "*", "Negotiated CAP capabilities: %s", msg);
+    strlcpy(cap.negotiated, msg, sizeof cap.negotiated);
+  } else if (!strcmp(cmd, "ACK")) {
+    putlog(LOG_MISC, "*", "%s acknowledged %s", from, msg);
+    strncat(cap.negotiated, msg, (sizeof cap.negotiated - strlen(cap.negotiated) - 1));
+  }
```
Comment (michaelortmann) [inline src/mod/server.mod/servmsg.c:1269]: maybe add:

else if (cmd[0])                                                             
  putlog(LOG_MISC, "*", "%s subcommand %s unknown, CAP END", from, cmd);

File: src/mod/server.mod/servmsg.c (hunk 4 above)
Comment (vanosg) [reply src/mod/server.mod/servmsg.c:1204]: We don't have an established coding guide... maybe we should, but it sounds like a lot of work :P I think we can make a case either way on this

File: src/mod/server.mod/servmsg.c [hunk 6]
```diff
@@ -22,13 +22,17 @@
 
 #include "../irc.mod/irc.h"
 #include "../channels.mod/channels.h"
+#include "server.h"
 
 static time_t last_ctcp = (time_t) 0L;
 static int count_ctcp = 0;
 static char altnick_char = 0;
+struct cap_list cap = {"", "", ""};
+char capes[64][32] = { 0 };
```
Comment (michaelortmann) [inline src/mod/server.mod/servmsg.c]: {0} seems to be c99 ok. but pls change it anyway to silence the following compiler warning:
clang -fPIC -g -O2 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/usr/include -g3 -DDEBUG -DDEBUG_ASSERT -DDEBUG_MEM -DDEBUG_DNS  -DMAKING_MODS -c .././server.mod/server.c && mv -f server.o ../
In file included from .././server.mod/server.c:128:
./servmsg.c:31:24: warning: suggest braces around initialization of subobject [-Wmissing-braces]
char capes[64][32] = { 0 };
                       ^
                       {}
1 warning generated.

**Pull request merged**